alexsg2/Practice_10k_Music_App
# Setting up redux example

DO NOT MERGE THIS!

This is just a hacky example to show how redux works.

Redux basically just lets us have a global object we can access from any component. This is useful because react's main way of handling state (useState) is local the component it is declared in. If another component needs to use that variable then it has to be passed down. If the component that needs the variable is not lower in the tree then the component the variable is declared in then you are kinda out of luck. There are ways to do so but redux the best option for us.


How to test:
Start the app

Go to the login page

The login button now has an integer (count) which is stored in the redux store. Clicking the login button increments the integer by one. Click it a few times.

Go back to the start screen and the go to the register page.

The register button is using count as well and it has the same value as the login page! 

Click the register button a few times. It decreases the count value by 1.

Navigating back to the login page. The value should be the same as the register page.

![IMG_9553](https://github.com/alexsg2/Practice_10k_Music_App/assets/25776618/0375eac8-2cb2-4de6-b239-8e458c17ae8c)
![IMG_9554](https://github.com/alexsg2/Practice_10k_Music_App/assets/25776618/5b14d396-a71e-4cb5-9607-cdc006aca9a3)
